### ISSUE TITLE
AttributeError: type object 'QEvent' has no attribute 'PinchGesture' on Apple laptop trackpad

### ISSUE THREAD

**Issue by kevinyamauchi** [napari/napari#3263]
## 🐛 Bug

When I make the pinch  gesture on my Apple laptop's trackpad (usually on accident), I receive the `type object 'QEvent' has no attribute 'PinchGesture` error below

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/opt/miniconda3/envs/napari-test/lib/python3.8/site-packages/vispy/app/backends/_qt.py in event(self=, ev=)
    572             self._vispy_canvas.events.touch(type='end')
    573         if t == qt_event_types.Gesture:
--> 574             gesture = ev.gesture(qt_event_types.PinchGesture)
        gesture = undefined
        ev.gesture = 
        qt_event_types.PinchGesture = undefined
    575             if gesture:
    576                 (x, y) = _get_qpoint_pos(gesture.centerPoint())

AttributeError: type object 'QEvent' has no attribute 'PinchGesture'
```

## To Reproduce

1. Open napari on a Mac laptop
2. Make a "pinch" (bring two fingers together on the trackpad) in the canvas

## Expected behavior

The gesture is silently ignored.

## Environment

Instalation:
```
# fresh python 3.8 conda env
pip install --pre 'napari[all]'
```
napari: 0.4.11rc2
Platform: macOS-10.15.7-x86_64-i386-64bit
System: MacOS 10.15.7
Python: 3.8.11 (default, Aug 6 2021, 08:56:27) [Clang 10.0.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.2
SciPy: 1.7.1
Dask: 2021.08.1
VisPy: 0.8.0

OpenGL:
- GL version: 2.1 INTEL-14.7.17
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1440x900, scale 2.0

Plugins:
- console: 0.0.3
- scikit-image: 0.4.11rc2
- svg: 0.1.5

**Comment by psobolewskiPhD** [comment]
Intestesting, I remember trying the pinch to zoom when I first started using napari. I didn't see any effect so abandoned it. 
For me it still does nothing and I don't get an error. Same behavior as 4.10.
I mean it would be cool if it worked though! (The trackpad/mouse gestures to zoom and pan in napari being so fluid and easy are one of the things that really sold me on it vs. ImageJ.)

Here's my napari details:
<details>
napari: 0.4.11rc2 
Platform: macOS-11.5.2-arm64-arm-64bit 
System: MacOS 11.5.2 
Python: 3.9.6 | packaged by conda-forge | (default, Jul 6 2021, 08:51:19) [Clang 11.1.0 ] 
Qt: 5.15.2 
PyQt5: 5.15.4 
NumPy: 1.21.0 
SciPy: 1.7.0 
Dask: 2021.06.2 
VisPy: 0.7.0 

OpenGL:
- GL version: 2.1 Metal - 71.7.1
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Plugins:
- 3 Utilities: 0.1.4
- 4 Measurements (skimage): 0.1.4
- StarDist: 2021.6.28
- aicsimageio-in-memory: 0.3.5
- aicsimageio-out-of-memory: 0.3.5
- animation: 0.0.2
- clEsperanto: 0.9.10
- console: 0.0.3
- ome-types: 0.2.6
- scikit-image: 0.4.11rc2
- svg: 0.1.5
- video
</details>

**Comment by kevinyamauchi** [comment]
Thanks for the feedback @psobolewskiPhD ! I think this is a vispy issue that was recently introduced. The difference might be that you have vispy 0.7.0 and I have vispy 0.8.0.

In light of that, the easy fix might be to pin vispy to `== 0.7.0`  or `< 0.8.0` for now.

**Comment by psobolewskiPhD** [comment]
I have to help a user with microscope today, but I'll try to clone my env and update vispy see what happens.

**Comment by kevinyamauchi** [comment]
Thanks!

**Comment by kevinyamauchi** [comment]
I had a quick minute, so I just tested with `vispy==0.7.0` and that seems to work for me. I think we need to just exclude `0.8.0` for napari `0.4.11`. Thoughts @sofroniewn ?

**Comment by djhoese** [comment]
I can make a 0.8.1 release today with the fix if someone can point me to why this doesn't work. On PyQt6 `QtCore.Qt.PinchGesture` exists so we have that correct. So what's going on with QEvent :thinking:

**Comment by djhoese** [comment]
@tlambert03 says that PyQt6 works for him after his latest related PR. On my non-mac system (PopOS) I don't even have the attribute that the backend is using:

```python
qt_event_types = QtCore.QEvent.Type if PYQT6_API else QtCore.QEvent
...
gesture = ev.gesture(qt_event_types.PinchGesture)
```

https://github.com/vispy/vispy/blob/e2bdc37457c8c5709218a7c9b2fbe8e218bd26a7/vispy/app/backends/_qt.py#L570

**Comment by djhoese** [comment]
Found the bug:

https://github.com/vispy/vispy/pull/2172/files#diff-211d803936a45285d93b8cbb6315dd40a722ef61afddfe3f1ba268951254577dR574

Edit: I'll fix this after waking up a bit more.

**Comment by tlambert03** [comment]
see this little bit here: https://www.riverbankcomputing.com/static/Docs/PyQt5/gotchas.html#enums

tldr:  starting with sip v4.19.9 (around PyQt 5.11) all enums are now scoped, and starting in PyQt6, the backwards compatibility allowing enums to be accessed in their top level namespace has been removed (so there may be a few more of these lurking around).
If you're still supporting < 5.11, you need to deal with this yourself (libs like qtpy can help) ... otherwise, the scoped variants can be used all over the place.  (i.e. prefer `Qt.GestureType.PinchGesture` over `Qt.PinchGesture` even if you're on `5.12`)

**Comment by kevinyamauchi** [comment]
> Found the bug:
> 
> https://github.com/vispy/vispy/pull/2172/files#diff-211d803936a45285d93b8cbb6315dd40a722ef61afddfe3f1ba268951254577dR574
> 
> Edit: I'll fix this after waking up a bit more.

Thank you, @djhoese ! I am happy to test on my mac when you make the PR.

**Comment by djhoese** [comment]
> libs like qtpy can help

When they support PySide6/PyQt6 PRs are welcome in vispy to switch to it :grin:

**Comment by djhoese** [comment]
This has been fixed as part of https://github.com/vispy/vispy/pull/2202 and is being released right now in vispy 0.8.1.

...and now builds are failing. It should be released today no matter what I have to do.

**Comment by djhoese** [comment]
VisPy 0.8.1 is out on PyPI now. Conda-forge package should be out later today.

**Comment by kevinyamauchi** [comment]
> VisPy 0.8.1 is out on PyPI now. Conda-forge package should be out later today.

Amazing! Thank you for the quick fix, @djhoese !

**Comment by kevinyamauchi** [comment]
@sofroniewn , do you think we should exclude vispy 0.8.0 from the requirements just to be safe or leave it as is. If it's the latter, we can close this issue.

**Comment by sofroniewn** [comment]
> do you think we should exclude vispy 0.8.0 from the requirements just to be safe or leave it as is. If it's the latter, we can close this issue.

I don't have strong feelings, maybe might as well make a PR to exclude. Would be a shame for someone to be on `0.8.0` when they could just as easily now be on `0.8.1` and have it work

**Comment by djhoese** [comment]
As a non-napari dev or contributor even, my opinion is that you shouldn't put the work in to limit against 0.8.0. This was a small bug that was only run into by accident for an event type that isn't even currently used by napari (from my understanding).

**Comment by kevinyamauchi** [comment]
Sorry, I think I wasn't super clear about how this manifests in napari. It could be that I am super clumsy, but when I am interacting with a 3D image using my trackpad, I accidentally make the "pinch" gesture fairly often. While we haven't hooked anything up to this gesture in napari, this raises the `AttributeError` described above (I'm not sure if we could do something with our implementation to prevent this). In light of this, I've made a small PR to exclude 0.8.0 for now.

Thanks again for the quick fix, @djhoese !
